src/Controller/CustomerController.php

```diff
@@ -2,21 +2,157 @@
 
 namespace App\Controller;
 
+use App\DTO\CustomerDTO;
+use App\Entity\Customer;
+use App\Entity\User;
 use App\Repository\CustomerRepository;
+use App\Repository\UserRepository;
+use Doctrine\ORM\EntityManager;
+use Doctrine\ORM\EntityManagerInterface;
+use JMS\Serializer\SerializationContext;
 use JMS\Serializer\SerializerInterface;
+use Psr\Cache\InvalidArgumentException;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\IsGranted;
 use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
 use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\PasswordHasher\Hasher\UserPasswordHasherInterface;
 use Symfony\Component\Routing\Annotation\Route;
+use Symfony\Component\Validator\Validator\ValidatorInterface;
+use OpenApi\Annotations as OA;
+use Symfony\Contracts\Cache\ItemInterface;
+use Symfony\Contracts\Cache\TagAwareCacheInterface;
+
 
 class CustomerController extends AbstractController
 {
-    #[Route('{id}/customers', name: 'app_user_customers')]
-    public function getAllCustomers(User $user, CustomerRepository $customerRepository, SerializerInterface $serializer): JsonResponse
+    /**
+     * Cette méthode retourne l'ensemble des clients liés à un utilisateur
+     * @param Request $request
+     * @param User $user
+     * @param CustomerRepository $customerRepository
+     * @param SerializerInterface $serializer
+     * @param TagAwareCacheInterface $cache
+     * @return JsonResponse
+     * @throws InvalidArgumentException
+     * @OA\Tag(name="Customer")
+     * * @OA\Parameter(
+     *     name="page",
+     *     in="query",
+     *     description="La page que l'on veut récupérer",
+     *     @OA\Schema(type="int")
+     * )
+     *
+     */
+    #[Route('/api/users/{id}/customers', name: 'app_user_customers', methods: ['GET'])]
+    public function getAllCustomers(Request $request, User $user, CustomerRepository $customerRepository, SerializerInterface $serializer, TagAwareCacheInterface $cache): JsonResponse
     {
-        $customers = $customerRepository->findBy(['user' => $user]);
-        $jsonCustomers = $serializer->serialize($customers, 'json');
+        $page = $request->get('page', 1);
+        $limit = $this->getParameter('app.limit_per_page_param');
+        $offset = (($page * $limit)-$page);
```

Comment:
You  should move this in your custom repo method and change the compute:

```suggestion
        $offset = (($page * $limit)-$limit);
```